### PR TITLE
block_hotplug: Refectoring block_hotplug

### PR DESCRIPTION
**Pull request by YongxueHong**
ID: 1550800, 1617971, 1624674
Signed-off-by: Yongxue Hong <yhong@redhat.com>

### CODE REVIEW

**Comment by YongxueHong** [comment]
Depends on https://github.com/avocado-framework/avocado-vt/pull/1908

**Comment by YongxueHong** [comment]
@CongLi , @peixiu @XueqiangWei @kimi1978wy @lolyu , help me review, thanks.

File: qemu/tests/block_hotplug.py [hunk 1]
```diff
@@ -21,144 +22,174 @@ def run(test, params, env):
...
-    img_list = params.get("images").split()
+    def check_unplugged_disks(session, disks_before_unplug):
+        """ Check the unplungged disks. """
+        unplug_disks = utils_misc.wait_for(
+            lambda: _get_new_disk(disks_before_unplug, _find_disks(session)),
+            pause)
+        if unplug_disks is None:
+            unplug_disks = set()
+        if len(unplug_disks) != extra_disks_num:
+            print('yhong--->unplug_disks: %s' % unplug_disks)
```
Comment (lolyu) [inline qemu/tests/block_hotplug.py]: redundant `print`.

File: qemu/tests/block_hotplug.py [hunk 2]
```diff
@@ -21,144 +22,174 @@ def run(test, params, env):
...
+        """ Create block devices. """
+        return vm.devices.images_define_by_params(
+            image_name, params.object_params(image_name), 'disk')
+
+    def get_block_devices(objs):
+        """ Get block devices. """
+        if isinstance(objs, str):
+            devs = [dev for dev in vm.devices if dev.get_param("id") == objs]
+        else:
+            devs = [dev for dev in objs if not isinstance(dev, qdevices.QDrive)]
```
Comment (lolyu) [inline qemu/tests/block_hotplug.py]: what if there is an iothread object in `objs`.

File: qemu/tests/block_hotplug.py (hunk 1 above)
Comment (YongxueHong) [reply qemu/tests/block_hotplug.py]: Sorry for this...

File: qemu/tests/block_hotplug.py [hunk 3]
```diff
@@ -21,144 +22,173 @@ def run(test, params, env):
...
 
-    img_list = params.get("images").split()
+    def check_unplugged_disks(session, disks_before_unplug):
+        """ Check the unplungged disks. """
+        unplug_disks = utils_misc.wait_for(
+            lambda: _get_new_disk(disks_before_unplug, _find_disks(session)),
+            pause)
+        if unplug_disks is None:
+            unplug_disks = set()
+        if len(unplug_disks) != extra_disks_num:
```
[the remark this answers is not in the document]
Comment (YongxueHong) [reply qemu/tests/block_hotplug.py]: Thanks, have updated.

**Comment by YongxueHong** [comment]
Closed it, and track by https://github.com/autotest/tp-qemu/pull/1720.

**Pull request closed without merging**